modules/service/src/main/scala/lucuma/odb/service/ItcService.scala

```diff
@@ -422,8 +422,8 @@ object ItcService {
         resultSet: AsterismResult
       )(using Transaction[F]): F[Unit] = {
 
-        def insertOrUpdateSingleTarget(acquisition: TargetResult)(science: TargetResult): F[Unit] = {
-          val h = hash((acquisition, science))
+        def insertOrUpdateSingleTarget(acquisition: TargetImagingResult)(science: TargetSpectroscopyResult): F[Unit] = {
+          val h = hash(acquisition.input, science.input)            
```

Comment:
Here is the caching issue (it was hashing the entire result rather than just the inputs). Took a while to find but easy to fix.

Reply:
Thanks for handling this.